### ISSUE TITLE
jwtOptionsProvider doesn't work

### ISSUE THREAD

**Issue by Spaeda** [auth0/angular2-jwt#695]
Hello,

I find many outdated posts about a similar issue but nothing fix it.

I use version 5.0.2.
`"@auth0/angular-jwt": "^5.0.2"`

I try to use interceptor to inject bearer token. I have 401 error because access token is not correctly retrieve.
My request only send this: `www-authenticate: Bearer`

When I debug my code, factory code is never call. So I add break point in `auth0-angular-jwt.js` to check it.
And in `forRoot` method, `options.config` is null.
```
static forRoot(options) {
        return {
            ngModule: JwtModule,
            providers: [
                {
                    provide: HTTP_INTERCEPTORS,
                    useClass: JwtInterceptor,
                    multi: true,
                },
                options.jwtOptionsProvider || {
                    provide: JWT_OPTIONS,
                    useValue: options.config,
                },
                JwtHelperService,
            ],
        };
    }
```

I also check `JwtHelperService` class and this line `this.tokenGetter = (config && config.tokenGetter) || function () { };` has also a null value for `config`.

I read documentation again and again but I don't understand my issue.

my app module:
```
import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { InjectionToken, NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { JwtModule, JWT_OPTIONS } from '@auth0/angular-jwt';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { CoreModule, PORTAL_API_URL } from './core/core.module';
import { AuthenticationService } from './core/services/authentication.service';
import { AuthenticationModule } from './modules/authentication/authentication.module';
import { SharedModule } from './shared/shared.module';

export function jwtOptionsFactory(authenticationService: AuthenticationService, portalApiUrl: InjectionToken<string>) {
  return {
    tokenGetter: () => {
      return authenticationService.accessToken;
    },
    allowedDomains: [
      "localhost:5011"
    ],
    throwNoTokenError: true,
  }
}

@NgModule({
  declarations: [AppComponent],
  imports: [
    CommonModule,
    SharedModule,
    AuthenticationModule,
    BrowserModule,
    AppRoutingModule,
    HttpClientModule,
    JwtModule.forRoot({
      jwtOptionsProvider: {
        provide: JWT_OPTIONS,
        useFactory: jwtOptionsFactory,
        deps: [AuthenticationService, PORTAL_API_URL],
      }
    }),
    CoreModule,
  ],
  providers: [AuthenticationService],
  bootstrap: [AppComponent],
})
export class AppModule {}
```

my authentication service:
```
@Injectable({ providedIn: 'root' })
export class AuthenticationService {
  private readonly storageKey: string = 'tokens';
  private storage: Storage = sessionStorage;

  private isAuthenticated: boolean = false;
  public status: Subject<boolean> = new Subject();

  private token: IAuthenticationToken | undefined = undefined;

  constructor(private authClient: AuthClient, private userService: UserService) {
    const storedToken = this.storage.getItem(this.storageKey);
    if (storedToken) {
      this.token = JSON.parse(storedToken);
      this.isAuthenticated = true;
    }
    this.status = new BehaviorSubject<boolean>(this.isAuthenticated);
  }

  public signIn(model: LoginViewModel) {
    return this.authClient.signIn(model)
      .subscribe((user) => {
        if (user) {
          if (user.tokens) {
            let accessToken = user.tokens.find(
              (o) => o.type == UserTokenType.Access
            );
            let refreshToken = user.tokens.find(
              (o) => o.type == UserTokenType.Refresh
            );
            if (accessToken?.value && refreshToken?.value) {
              this.setTokens(accessToken.value, refreshToken.value);
              this.isAuthenticated = true;
              this.status.next(this.isAuthenticated);
            }
          }
          this.userService.setCurrentUser(user);
          // Redirect to home page
        }
      });
  }

  get accessToken(): string | undefined {
    return this.token?.accessToken;
  }
}
```

I need this package for `JwtHelperService` so I don't want use a custom interceptor :s

Can you help me pls? I become crazy with this issue.

PS: my authentication service correctly find an return access token.

**Comment by dbrils** [comment]
I got the same issue, after updating to `5.0.2` from `4.1.2` (I was migrating my Angular 8 app to the most recent version) I stumbled across the same issue. The package is no longer adding the JWT access token to the requests. I double checked my implementation, but it's exactly the same as Ionic example: https://github.com/auth0/angular2-jwt#configuration-for-ionic-2.

Could someone have a look at this issue? The only solution I see it writing my own auth token interceptor.
